### PR TITLE
[Feature] Earnings

### PR DESCRIPTION
**Pull request by bog5651**

### CODE REVIEW

File: src/store/modules/earnings/modules/olympus/actions.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+      if (rootState.account?.networkInfo === undefined) {
+        throw new Error('failed to get network info');
+      }
+
+      if (rootState.account?.provider === undefined) {
+        throw new Error('failed to get provider');
+      }
+
+      commit('setOlympusAPY', '7.333');
```
Comment (Sinu5oid) [inline src/store/modules/earnings/modules/olympus/actions.ts:40]: olympus APY would be contract interaction (as balance and others)

File: src/store/modules/earnings/modules/olympus/actions.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      const olympusData = await getOlympusData(
+        rootState.account.currentAddress,
+        rootState.account.networkInfo.network,
+        rootState.account.provider.web3
+      );
+      commit('setOlympusBalance', olympusData.balance);
+
+      await dispatch('fetchOlympusInfo');
+      await dispatch('fetchOlympusPriceInWeth');
```
Comment (Sinu5oid) [inline src/store/modules/earnings/modules/olympus/actions.ts:49]: You can await for `Promise.all` of them

File: src/store/modules/earnings/modules/olympus/mutations.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  setIsLoading(state, isLoading): void {
+    state.isLoading = isLoading;
+  },
+  setOlympusAPY(state, apy): void {
+    state.olympusAPY = apy;
+  },
+  setOlympusBalance(state, balance): void {
+    state.olympusBalance = balance;
+  },
+  setOlympusInfo(state, info): void {
```
Comment (Sinu5oid) [inline src/store/modules/earnings/modules/olympus/mutations.ts:15]: missing argument types here and everywhere else. Also info could be `undefined` as stated in types.ts

File: src/styles/_404.less [hunk 4]
```diff
@@ -1,3 +1,5 @@
+@import './_variables.less';
```
Comment (Sinu5oid) [inline src/styles/_404.less:1]:
```suggestion
@import (reference) './_variables.less';
```

`(reference)` ensures that this piece of declared import is not included to the final compiled css bundle unless it's necessary

File: src/styles/_execute_modal.less [hunk 5]
```diff
@@ -1,3 +1,5 @@
+@import './_variables.less';
```
Comment (Sinu5oid) [inline src/styles/_execute_modal.less:1]: Same here, `(reference)`

File: src/views/earnings/ethereum/earnings-ethereum-manage.vue [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<template>
+  <earnings-ethereum-manage-empty v-if="!hasActiveEarnings" />
```
Comment (Sinu5oid) [inline src/views/earnings/ethereum/earnings-ethereum-manage.vue:2]: I believe there should be also an active component

File: src/views/earnings/ethereum/earnings-ethereum-manage.vue [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { mapGetters } from 'vuex';
+
+import EarningsEthereumManageEmpty from '@/components/earnings/ethereum/earnings-ethereum-manage-empty.vue';
```
Comment (Sinu5oid) [inline src/views/earnings/ethereum/earnings-ethereum-manage.vue:9]: No way, let's utilize the index.ts in `@/components/earnings/ethereum`

File: src/views/earnings/ethereum/earnings-ethereum-withdraw.vue [hunk 8]
```diff
@@ -0,0 +1,11 @@
+<template>
+  <div></div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+  name: 'EarningsEthereumWithdraw'
```
Comment (Sinu5oid) [inline src/views/earnings/ethereum/earnings-ethereum-withdraw.vue:9]: If this component is unused then it's better to remove it or populate it with some code anyway

File: src/views/earnings/olympus/earnings-olympus-global-analytics.vue [hunk 9]
```diff
@@ -0,0 +1,85 @@
+<template>
+  <secondary-page has-back-button hide-title @back="handleBack">
+    <div>
+      <secondary-page-simple-title
+        class="page-title"
+        :description="$t('earnings.txtOverview')"
+        :title="$t('earnings.lblOverview', { token: 'Olympus ' })"
```
Comment (Sinu5oid) [inline src/views/earnings/olympus/earnings-olympus-global-analytics.vue:7]: Typo

File: src/views/earnings/olympus/earnings-olympus-global-analytics.vue [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+          :description="$t('earnings.lblEstimatedEarningsTomorrow')"
+          :value="0"
+        />
+        <statement-list-item
+          :description="$t('earnings.lblEstimatedEarningsThisMonth')"
+          :value="0"
+        />
+        <statement-list-item
+          :description="$t('earnings.lblEstimatedEarningsAnnually')"
+          :value="0"
```
Comment (Sinu5oid) [inline src/views/earnings/olympus/earnings-olympus-global-analytics.vue:52]: let's move values to the appropriate computed entries

File: src/views/earnings/olympus/earnings-olympus-monthly-statistics.vue [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+    averageDailyEarnings(): string {
+      const value = formatToNative(0);
+      return `${getSignIfNeeded(value, '+')}$${value}`;
+    }
+  },
+  async mounted() {
+    // await this.fetchMonthlyStats({
+    //   year: pageDate.get('year'),
+    //   month: pageDate.get('month') + 1
+    // } as SavingsGetReceiptPayload);
```
Comment (Sinu5oid) [inline src/views/earnings/olympus/earnings-olympus-monthly-statistics.vue:127]: Mock this with an appropriate action

**Review by Sinu5oid — COMMENTED** [review]
Good job! Let's fix everything we've found so far, toggle feature-flags off and merge it to the development
Should be pretty much safe as the feature is under feature flag

File: src/views/earnings/olympus/earnings-olympus-global-analytics.vue (hunk 9 above)
Comment (bog5651) [reply src/views/earnings/olympus/earnings-olympus-global-analytics.vue]: Can u write more detailed?

Comment (Sinu5oid) [reply src/views/earnings/olympus/earnings-olympus-global-analytics.vue]: There is a typo `Olympus ` -> `Olympus`

**Review by Sinu5oid — APPROVED** [review]
Please disable the feature flag and LGTM! :rocket:

**Pull request merged**